src/governance.cpp

```diff
@@ -147,6 +147,11 @@ void CGovernanceManager::ProcessMessage(CNode* pfrom, std::string& strCommand, C
         uint256 nHash = govobj.GetHash();
         std::string strHash = nHash.ToString();
 
+        LOCK2(cs_main, cs);
+
+        pfrom->setAskFor.erase(nHash);
+        mapAlreadyAskedFor.erase(nHash);
```

Comment:
Good catch 👍 I haven't thought of it, not sure why there is one for txes as well, might be some legacy stuff.